docs/tools/platformio.mdx

```diff
@@ -0,0 +1,128 @@
+framework = stm32cube
+
+build_unflags = -Os
+build_flags =
+    -include node_config.h
+    -O1
+lib_deps =
+    luos_engine
+```
+
+This project in example is Luos engine dependent. PlatformIO will download from its registry, the last version of the library Luos engine .The file "node_config.h" allowing user to configure Luos parameter and hardware definition to be add to the project.
+
+Building project exemple:
+<div align="center">
+  <Image src="/img/telemetry-1.svg" darkSrc="/img/telemetry-1.svg" />
+</div>
+
+You must define the <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow"> HAL family <IconExternalLink width="10"></IconExternalLink> </a> corresponding to your MCU. see the [compatibility page](/docs/compatibility/mcu_demoboard)
+
+Add the definition of the HAL you want to use as follow:
+
+Platformio.ini project example:
+```AsciiDoc
+[env:nucleo_f072rb]
+platform = ststm32
+board = nucleo_f072rb
+framework = stm32cube
+
+build_unflags = -Os
+build_flags =
+    -include node_config.h
+    -DLUOSHAL=STM32F0
+    -O1
+lib_deps =
+    luos_engine
+```
+
+
+Building project exemple:
+<div align="center">
+  <Image src="/img/telemetry.svg" darkSrc="/img/telemetry.svg" />
+</div>
+
+By default the telemetry is enable and send anonymous information to luos (only for Luos engine experience improving). To disable place the flag NOTELEMETRY in 'Platformio.ini' the same way as LUOSHAL Flag
+
+###  Definition for gate tool and pipe
+
+The same way for tools, some pre-compile definition may be use to configure Luos tools when these add in your project.
+For more information, see the script select_format_script.py in <a href="https://github.com/Luos-io/luos_engine/tree/main/tool_services/gate" target="_blank" rel="external nofollow"> gate tool <IconExternalLink width="10"></IconExternalLink> </a> and select_hal_script.py in <a href="https://github.com/Luos-io/luos_engine/tree/main/tool_services/pipe" target="_blank" rel="external nofollow"> pipe <IconExternalLink width="10"></IconExternalLink> </a>
+
+|      Definition       |                          Description                        |         Value         |
+| :-------------------: |-----------------------------------------------------------: | :-------------------: |
+|      GATEFORMAT       | Chose the format of data send by the gate to through a pipe |       TinyJson        |
+
+|      Definition       |                                      Description                                   |         Value         |
+| :-------------------: |----------------------------------------------------------------------------------: | :-------------------: |
+|       PIPEMODE        | Choose Pipe style to send or receive information through a specific physical Layer |    SERIAL/WIFI/BLE    |
+|       PIPEHAL         |             Choose Pipe HAL match a specific physical Layer with the board         |      see pipe HAL     |
+
```

Comment:
Same for values explain the folder thing, and add example.

Reply:
done